javatests/build/bazel/tests/integration/BazelBaseTestCaseTest.java

```diff
@@ -49,28 +47,26 @@ public void testRunningBazelInGivenRelativeDirectory() throws Exception {
     assertThat(cmd.outputLines()).contains("in bar");
   }
 
+  /**
+   * Write out a BUILD file that loads the `bazel_java_integration_test` rule.
+   *
+   * This is a demo of how to load a .bzl file that is included in the parent repo.
+   * The test itself doesn't invoke Bazel, it's just a wrapper.
+   */
   @Test
   public void testTestSuiteExists() throws Exception {
     loadIntegrationTestRuleIntoWorkspace();
-    setupPassingTest("IntegrationTestSuiteTest");
-
-    driver.bazel("test", "//:IntegrationTestSuiteTest").mustRunSuccessfully();
-  }
-
-  @Test
-  public void testJvmFlags() {
-    org.hamcrest.MatcherAssert.assertThat(System.getProperty("foo.bar"), is("true"));
-  }
-
-  private void setupPassingTest(final String testName) throws IOException {
-    writePassingTestJavaSource(testName);
-    writeTestBuildFile(testName);
-  }
-
-  private void writeTestBuildFile(final String testName) throws IOException {
+    final String testName = "IntegrationTestSuiteTest";
```

Comment:
I have to be honest and say I feel these were not indirection but abstraction.
The current state feels, to me, as an overload of details and literals.
I'd really like for us to keep the previous state.